wagtail/contrib/styleguide/views.py

```diff
@@ -95,58 +96,65 @@ def __init__(self, *args, **kwargs):
-        request,
-        _("Error message"),
-        buttons=[messages.button("", _("View live")), messages.button("", _("Edit"))],
-    )
-
-    paginator = Paginator(list(range(100)), 10)
-    page = paginator.page(2)
-
-    icon_hooks = hooks.get_hooks("register_icons")
-    registered_icons = itertools.chain.from_iterable(hook([]) for hook in icon_hooks)
-    all_icons = defaultdict(list)
-    for icon_path in registered_icons:
-        folder, filename = os.path.split(icon_path)
-        icon = render_to_string(icon_path)
-        id_match = icon_id_pattern.search(icon)
-        name = id_match.group(1) if id_match else None
-        source_match = icon_comment_pattern.search(icon)
-
-        all_icons[folder].append(
-            {
-                "folder": folder,
-                "file_path": icon_path,
-                "name": name,
-                "source": source_match.group(1) if source_match else None,
-                "icon": icon,
-            }
+        messages.success(
+            self.request,
+            _("Success message"),
+            buttons=[
+                messages.button("", _("View live")),
+                messages.button("", _("Edit")),
+            ],
+        )
+        messages.warning(
+            self.request,
+            _("Warning message"),
+            buttons=[
+                messages.button("", _("View live")),
+                messages.button("", _("Edit")),
+            ],
+        )
+        messages.error(
+            self.request,
+            _("Error message"),
+            buttons=[
+                messages.button("", _("View live")),
+                messages.button("", _("Edit")),
+            ],
         )
 
-    return TemplateResponse(
-        request,
-        "wagtailstyleguide/base.html",
-        {
-            "all_icons": all_icons.items(),
-            "search_form": form,
-            "example_form": example_form,
-            "example_page": page,
-        },
-    )
+        paginator = Paginator(list(range(100)), 10)
+        context["example_page"] = paginator.page(2)
+
+        icon_hooks = hooks.get_hooks("register_icons")
+        registered_icons = itertools.chain.from_iterable(
+            hook([]) for hook in icon_hooks
+        )
+        all_icons = defaultdict(list)
+        for icon_path in registered_icons:
+            folder, filename = os.path.split(icon_path)
+            icon = render_to_string(icon_path)
+            id_match = icon_id_pattern.search(icon)
+            name = id_match.group(1) if id_match else None
+            source_match = icon_comment_pattern.search(icon)
+
+            all_icons[folder].append(
+                {
+                    "folder": folder,
+                    "file_path": icon_path,
+                    "name": name,
+                    "source": source_match.group(1) if source_match else None,
+                    "icon": icon,
+                }
+            )
+
+        context["all_icons"] = all_icons.items()
```

Comment:
It might be nicer to pull out the icons generation to a class method if that's ok.


Something like this.

```py
    def get_icons(self):
        icon_hooks = hooks.get_hooks("register_icons")
        registered_icons = itertools.chain.from_iterable(
            hook([]) for hook in icon_hooks
        )
        all_icons = defaultdict(list)
        for icon_path in registered_icons:
            folder, filename = os.path.split(icon_path)
            icon = render_to_string(icon_path)
            id_match = icon_id_pattern.search(icon)
            name = id_match.group(1) if id_match else None
            source_match = icon_comment_pattern.search(icon)

            all_icons[folder].append(
                {
                    "folder": folder,
                    "file_path": icon_path,
                    "name": name,
                    "source": source_match.group(1) if source_match else None,
                    "icon": icon,
                }
            )
        return all_icons.items()
```

Then

```suggestion
        context["all_icons"] = self.get_icons()
```